src/ansys/fluent/core/streaming_services/events_streaming.py

```diff
@@ -1,30 +1,56 @@
 """Module for events management."""
 
+from enum import Enum
 from functools import partial
 import logging
-from typing import Callable, List
+from typing import Callable
 
 from ansys.api.fluent.v0 import events_pb2 as EventsProtoModule
-from ansys.fluent.core.exceptions import DisallowedValuesError, InvalidArgument
+from ansys.fluent.core.exceptions import InvalidArgument
 from ansys.fluent.core.streaming_services.streaming import StreamingService
 
 network_logger = logging.getLogger("pyfluent.networking")
 
 
-class EventsManager(StreamingService):
-    """Manages server-side events.
+class Event(Enum):
+    """Enumerates over supported server (Fluent) events."""
+
+    TIMESTEP_STARTED = "TimestepStartedEvent"
+    TIMESTEP_ENDED = "TimestepEndedEvent"
+    ITERATION_STARTED = "IterationStartedEvent"
+    ITERATION_ENDED = "IterationEndedEvent"
+    CALCULATIONS_STARTED = "CalculationsStartedEvent"
+    CALCULATIONS_ENDED = "CalculationsEndedEvent"
+    CALCULATIONS_PAUSED = "CalculationsPausedEvent"
+    CALCULATIONS_RESUMED = "CalculationsResumedEvent"
+    ABOUT_TO_LOAD_CASE = "AboutToReadCaseEvent"
+    CASE_LOADED = "CaseReadEvent"
+    ABOUT_TO_LOAD_DATA = "AboutToReadDataEvent"
+    DATA_LOADED = "DataReadEvent"
+    ABOUT_TO_INITIALIZE_SOLUTION = "AboutToInitializeEvent"
+    SOLUTION_INITIALIZED = "InitializedEvent"
+    REPORT_DEFINITION_CHANGED = "ReportDefinitionChangedEvent"
+    PLOT_SET_CHANGED = "PlotSetChangedEvent"
+    RESIDUAL_PLOT_CHANGED = "ResidualPlotChangedEvent"
+    SETTINGS_CLEARED = "ClearSettingsDoneEvent"
+    SOLUTION_PAUSE = "AutoPauseEvent"
+    PROGRESS_UPDATED = "ProgressEvent"
+    SOLVER_TIME_ESTIMATE_UPDATED = "SolverTimeEstimateEvent"
+    ERROR = "ErrorEvent"
+    COMMAND_COMPLETED = "CommandCompletedEvent"
+
+    @classmethod
+    def _missing_(cls, value: str):
```

Comment:
This should through if `value` doesn't match with any `cls` member.

Reply:
Done, Thank you.